docs/shared/recipes/tips-n-tricks/wait-for-tasks.md

```diff
@@ -0,0 +1,74 @@
+# Wait for Tasks to Finish
```

Comment:
Hmm...I'm not 100% convinced about this one yet. Because the `dependsOn` and `waitUntilTargets` kinda fulfill different purposes. I see the "Waiting for a Task" concept but..

1) `dependsOn` just runs tasks before running another one, establishing a " task pipeline"
2) `waitUntilTargets` is for long-running tasks and doesn't really wait until it finishes, but until its up and running

For 1) we already have our docs about the task pipeline. For 2) we don't really have anything useful and there's indeed a gap in our docs/nx implementation; I've been just chatting about this with @ZackDeRose yesterday.

So I'd focus the recipe on the long-running task part and ignore the `dependsOn` (maybe just mention it and then point to the other recipe/task running docs for those that accidentally landed on this recipe but rather want a task pipeline). On the long-running task story I think there are these use cases that come to my mind:

- spinning up multiple long-running targets (like serving multiple apps, spinning up frontend & backend API) where order doesn't necessarily matter => we can use run commands and the `parallel` flag (not sure whether we have a good recipe that describes that)
- spinning up multiple long-running targets where order matters

For the latter for instance a common use case can be if you want to serve the app and then run some task against it (e.g. e2e etc..). This is what also @ZackDeRose brought up. We have some executors that have this baked in, where you can provide a serve or build target and internally they'll make sure to launch it and wait, then continue (e.g. the Cypress one). The `waitUntilTargets` kinda goes in that direction. Ideally `dependsOn` would have something baked in that detects whether an executor is long-running etc..

Reply:
Removed the dependsOn section from the recipe and moved it under the node recipes.